### ISSUE TITLE
Check for existing episode locally before downloading another version

### ISSUE THREAD

**Issue by myndzero** [erengy/taiga#1004]
So I noticed this program tends to download multiple of the same episode which makes listing a preference order for subs kinda pointless (also downloads other ones not listed at all??).

As far as I can tell there is no way to add a filter that makes it check your local folders to see if there's already an episode in there before continuing on to download multiple of the same episode. Can we get this functionality added?

**Comment by paulo27ms** [comment]
That's one of the default filters...
If you don't want it to download multiple things you need to set preferences and filters.

**Comment by myndzero** [comment]
Can you screenshot me that then, but my defaults should also be there so I'm not sure why it wouldn't work properly?

**Comment by paulo27ms** [comment]
![image](https://user-images.githubusercontent.com/13546823/118322613-54f20100-b4f7-11eb-9a95-01b005b181a1.png)

**Comment by myndzero** [comment]
Mines the same so I'm not sure why, my setting area looks a tad different to yours though? 

![image](https://user-images.githubusercontent.com/84202322/118364502-04a09f00-b5dc-11eb-8cfa-77edc18a6e1f.png)

**Comment by paulo27ms** [comment]
Because you're on the stable and not beta.
But that won't affect the filters.
Next step is making sure Taiga is actually recognizing your episodes and your library is added with the correct path.

**Comment by myndzero** [comment]
Well I've got E:\Anime listed as a library with real time monitor checked there, and that's also where I've put the download location under torrents & checked to create subfolders using anime title name, 

This is my recognition page which looks fine to me. 
![image](https://user-images.githubusercontent.com/84202322/118389503-93a9c780-b66d-11eb-9c41-481c057ffabd.png)

**Comment by paulo27ms** [comment]
And in your list you can see series highlighted?

**Comment by myndzero** [comment]
Yes, but it's only like 2, so it's not grabbing majority of them nor is it sticking to the fansub group I put a filter for, it'll grab random ones still.

**Comment by paulo27ms** [comment]
I mean in your library, is it actually showing series highlighted in blue (i.e. with episodes recognized)?

**Comment by myndzero** [comment]
Yes? here's a pic 
![image](https://user-images.githubusercontent.com/84202322/118408012-ecf21500-b6c6-11eb-82f4-34bb7c1b72ad.png)

**Comment by paulo27ms** [comment]
This shows you only have those 2 series with episodes so of course it'll download anything new.
Do you actually have episodes for the rest?
When you play an episode are they actually recognized?
Have you tried forcing a refresh (F5)?

**Comment by myndzero** [comment]
That's the thing though, I don't have existing episodes for those 2 highlighted lol. I feel like my multiple download things is likely just related to the subs filter if anything. E.g megalo new ep that just came out it downloads like 5 different ones, whereas realistically it should only download one since I've only got one sub group set currently. 

I might change this filter and approach it from the other end... if it doesn't contain then drop?

**Comment by paulo27ms** [comment]
If there's no episodes in your library and they aren't watched then obviously it'll download everything it can, it's what you're setting it to do.
It'll only download what's uploaded, it doesn't search for old things to download just because a new episode is out.

**Comment by myndzero** [comment]
That doesn't explain why it's downloading multiple, why can't it just doesn't *one* of the same episode number?
It doesn't explain why megalo, which has the previous episode, was watched, still did the exact same thing as above when the next episode came out. 

I don't want it to download old stuff, I'm just trying to get it to download one copy of an episode, not every copy of the same episode by different sub groups. If that's what I'm setting it to do, how do I set it to not do that then?

So far all I'm seeing is this software is extremely flawed if it can't seemingly work this one aspect out.

**Comment by paulo27ms** [comment]
There's no issue in the filters as they can do exactly what you want them to do.
If you're downloading multiple copies of an episodes you need to:
- Set preference filters so only one type of file is selected per episode
- Set filters to exclude things you don't want
Else Taiga will always download everything it can because that's what you're telling it to do.

**Comment by myndzero** [comment]
Can you give me a filter screenshot example them for the one type per episode?

**Comment by paulo27ms** [comment]
For one, create a general filter for the fansub group you prefer.
What this does is:
- When there's multiple versions of the same episode present, **IF** SubsPlease is present, it only selects SubsPlease
- Hides everything else related from view
- Since you have the filter to hide downloaded episodes enable, you won't see more of the same episode (unless you delete it and don't mark it as watched)
- This obviously has the issue that SubsPlease isn't present for every show but obviously it's up to you to tell Taiga to not download basically everything in your watching list, add some sanity filters
 
![image](https://user-images.githubusercontent.com/13546823/118418265-b0ff8580-b6af-11eb-9736-a6a1a08cc4c5.png)

I'd advise for you to actually explore the filter options a bit and trying to figure out what you want to do, disable automatic downloads and as long as you don't press this button 
![image](https://user-images.githubusercontent.com/13546823/118418451-5d416c00-b6b0-11eb-9e9e-d4366e0f7113.png) you don't have to wait to see your new filters in action and what they are doing.

**Comment by myndzero** [comment]
Dunno hey. Went through and downloaded a bunch so I guess I could thoroughly test it incase it was screwing up due to lack of files, but it still downloaded duplicate episodes. 

Reading your comment I started to wonder if my discard downloaded episodes isn't it (it was default?) correct so I added if episode availability = true then discard/hide. I'm assuming this means local files and on the search list looks like it functions that way so hopefully? 

I think I need to move my hourly check back a bit thinking about it, I could see a world where it's grabbing the lower-down-the-list subs preferences based on timing and higher time between checks would probably fix that. I'm thinking it's finally all good and setup now but I'll just have to wait and see to be sure.
